tests/test_blank_footprint.py

```diff
@@ -0,0 +1,221 @@
+      return sim
+
+def test_outputs_to_log_no_blank(tmpdir):
+    """Check that logging from Consumables is as expected."""
+    intrinsic_availability = {0: 1.0, 1: 0.0}
+
+    sim = get_sim_with_dummy_module_registered(
+        data=create_dummy_data_for_cons_availability(
+            intrinsic_availability=intrinsic_availability,
+            months=[1],
+            facility_ids=[0]),
+        tmpdir=tmpdir,
+        run=False
+    )
+
+   # Edit the `initialise_simulation` method of DummyModule so that, during the simulation, an HSI is run.
+    def schedule_hsi(sim):
+        """Drop-in replacement for `initialise_simulation` in the DummyModule module."""
+        # Make the district for person_id=0 such that the HSI will be served by facility_id=0
+        sim = set_person_district_id(sim)
+
+        # Schedule the HSI event with a blank footprint for person_id=0
+        sim.modules['HealthSystem'].schedule_hsi_event(
+            hsi_event=get_dummy_hsi_event_instance(module=sim.modules['DummyModule'], facility_id=0, blank = blank_footprint),
+            topen=sim.start_date,
+            tclose=None,
+            priority=0
+        )
+
+
+    sim.modules['DummyModule'].initialise_simulation = schedule_hsi
+
+     # Simulate for one day
+    sim.simulate(end_date=sim.start_date + pd.DateOffset(days=1))
+
+     # Check that log is created and the content is as expected.
+    hsi_log = parse_log_file(sim.log_filepath)['tlo.methods.healthsystem.summary']['hsi_event_details']
+    assert hsi_log.iloc[0,1]['0']['appt_footprint'] == []
+
+
+
+
+def test_outputs_to_log_blank(tmpdir):
+    """Check that logging from Consumables is as expected."""
+    intrinsic_availability = {0: 1.0, 1: 0.0}
+
+    sim = get_sim_with_dummy_module_registered(
+        data=create_dummy_data_for_cons_availability(
+            intrinsic_availability=intrinsic_availability,
+            months=[1],
+            facility_ids=[0]),
+        tmpdir=tmpdir,
+        run=False
+    )
+
+    # Edit the `initialise_simulation` method of DummyModule so that, during the simulation, an HSI is run.
+    def schedule_hsi(sim):
+        """Drop-in replacement for `initialise_simulation` in the DummyModule module."""
+        # Make the district for person_id=0 such that the HSI will be served by facility_id=0
+        sim = set_person_district_id(sim)
+
+        # Schedule the HSI event with no blank footprint for person_id=0
+        sim.modules['HealthSystem'].schedule_hsi_event(
+            hsi_event=get_dummy_hsi_event_instance_blank_footprint(module=sim.modules['DummyModule'], facility_id=0),
+            topen=sim.start_date,
+            tclose=None,
+            priority=0
+        )
+
+    sim.modules['DummyModule'].initialise_simulation = schedule_hsi
+
+    # Simulate for one day
+    sim.simulate(end_date=sim.start_date + pd.DateOffset(days=1))
+
+    # Check that log is created and the content is as expected.
```

Comment:
I think we want to check that an HSI that has a blank footprint:
* does appear in the log key `hsi_event_details`
* does NOT appear in the log key `hsi_event_no_blank_footprint`